File: snippets/node/40-users-groups/02-user-detail.mjs
// import the shared client:  import { connect } from "elo-playground";
// topic:    Read a user's full record
// category: Users & groups
// id:       users.user-detail

import { connect } from "elo-playground";

// --- local ELO test box (override with ELOPG_* env vars or a .env) ---
const ELO_BASE_URL = process.env.ELOPG_ELO_BASE_URL || "http://localhost:9090/ix-Repository1";
const ELO_USER = process.env.ELOPG_ELO_USER || "Administrator";
const ELO_PASS = process.env.ELOPG_ELO_PASSWORD || "";

const elo = await connect({ baseUrl: ELO_BASE_URL, user: ELO_USER, password: ELO_PASS });

const res = await elo.call("checkoutUsers", {
  ids: [0, 12],                        // Administrator + Max Muster
  checkoutUsersZ: { bset: "513" },     // CHECKOUT_USERS.BY_IDS
});
const users = Array.isArray(res) ? res : res.users || [];

for (const u of users) {
  console.log(`${u.name}:`);
  console.log(`    groups     : ${JSON.stringify(u.groupList)}`);
  console.log(`    last login : ${u.lastLoginIso || "-"}`);
  console.log(`    flags      : ${u.flags}`);
}

File: snippets/python/40-users-groups/02-user-detail.py
# import the shared client:  from elo_playground import connect
# topic:    Read a user's full record
# category: Users & groups
# id:       users.user-detail

from elo_playground import connect

elo = connect()

res = elo.call("checkoutUsers", {
    "ids": [0, 12],                       # Administrator + Max Muster
    "checkoutUsersZ": {"bset": "513"},    # CHECKOUT_USERS.BY_IDS
})
users = res if isinstance(res, list) else res.get("users", [])

for u in users:
    print(f"{u['name']}:")
    print(f"    groups     : {u.get('groupList')}")
    print(f"    last login : {u.get('lastLoginIso', '-')}")
    print(f"    flags      : {u.get('flags')}")
